security.go

```diff
@@ -54,6 +57,10 @@ func NewSecurity() Security {
 		panic(err)
 	}
 
+	return NewSecurityWithKey(key)
+}
+
+func NewSecurityWithKey(key *ecdsa.PrivateKey) Security {
```

Comment:
Hmmm. We've been doing this with an options pattern for basically everything else. i.e Engine, Server, Route. I kinda think we should take the same approach here. So like the NewSecurity func would be this

```golang
// An SecurityOption represents configures the behavior of the Security struct.
type SecurityOption = func(*Security)

// NewEngine creates a new Engine with the given options.
// For example:
//
//	engine := fuego.NewSecurity(
//      WithKey(key)
// )
//
// Options all begin with `With`.
func NewSecurity(options ...SecurityOption) *Security { ... }
```